tls/s2n_fingerprint.c

```diff
@@ -112,11 +112,12 @@ static S2N_RESULT s2n_fingerprint_write_version(struct s2n_client_hello *ch,
         struct s2n_stuffer *output, struct s2n_hash_state *hash)
 {
     RESULT_ENSURE_REF(ch);
+
+    uint8_t high = (ch->version / 10);
+    uint8_t low = (ch->version % 10);
+    uint16_t version = (high << 8) | low;
+
     bool is_list = false;
-    uint16_t version = 0;
-    struct s2n_stuffer message = { 0 };
-    RESULT_GUARD_POSIX(s2n_stuffer_init_written(&message, &ch->raw_message));
-    RESULT_GUARD_POSIX(s2n_stuffer_read_uint16(&message, &version));
```

Comment:
I'm torn on whether this is an improvement :\ Math is definitely faster, but the stuffer operations were probably more readable.

Reply:
Yeah, true. The stuffers do seem more readable... Although we do this conversion in other places, so maybe it makes sense here. I could see it either way I guess.

Reply:
I'm in favor of the more efficient math. But we may be able to maintain readability: If we do the conversion in other places does it make sense to turn it into a macro/function to make it more readable?

Reply:
I think part of my hesitation to make this change is that it actually still wouldn't work for sslv2. In sslv2, we'd need to know that the ClientHello was an sslv2 ClientHello, but the version we'd include in the fingerprint wouldn't be sslv2. It'd be the conn->client_protocol version / the version listed in the ClientHello: basically the version we'd read from the stuffer, if s2n-tls didn't weirdly drop the first few bytes of the sslv2 ClientHello.